Eastern-Research-Group/mywaterway
# Feature/307 Add HUC boundaries layer to layer list

## Related Issues:
* [HMW-307](https://jira.epa.gov/secure/RapidBoard.jspa?rapidView=864&view=detail&selectedIssue=HMW-307)

## Main Changes:
* Added the HUC boundaries layer and the search icon layer to a group layer.
* Enabled layer list visibility for the group layer.
* Enabled the ability to toggle the search icon and boundaries sublayers separately.

## Steps To Test:
1. Confirm that the "Current Location" layer appears in the layer list for all Community page tabs
2. Confirm that the boundary layer is not initially visible by default on the "Who provides the drinking water here" sub-tab of the Drinking Water tab
3. Confirm that the HUC boundary shading is still removed when the Watershed Health Scores layer is visible
4. Test that the legend is properly updated when the group layer and child layers are toggled
